vector/src/main/java/im/vector/app/features/onboarding/OnboardingViewModel.kt

```diff
@@ -164,6 +165,7 @@ class OnboardingViewModel @AssistedInject constructor(
             is OnboardingAction.ProfilePictureSelected     -> handleProfilePictureSelected(action)
             OnboardingAction.SaveSelectedProfilePicture    -> updateProfilePicture()
             is OnboardingAction.PostViewEvent              -> _viewEvents.post(action.viewEvent)
+            OnboardingAction.StopEmailValidationCheck      -> currentJob = null
```

Comment:
as part of the registration flow it's cancelling the _wait for email verification_ step (when we send an email and wait for the user to click the verification link in their email client), however like you mentioned the viewmodel reuses the same `currentJob` instance for all of its tasks 


for some extra context 

- the currentJob uses a property setter to cancel any previous jobs on update

```kotlin
private var currentJob: Job? = null
    set(value) {
        // Cancel any previous Job
        field?.cancel()
        field = value
    }
```

The email validation waiting is tied to the pause/resume lifecycle 

```kotlin
override fun onResume() {
    viewModel.handle(OnboardingAction.PostRegisterAction(RegisterAction.CheckIfEmailHasBeenValidated(0)))
}

override fun onPause() {
    viewModel.handle(OnboardingAction.StopEmailValidationCheck)
}
```

I would prefer to avoid adding a comment if a suitable method could be extracted, how do you feel about...

```kotlin
OnboardingAction.StopEmailValidationCheck      -> cancelWaitForEmailValidation()

fun cancelWaitForEmailValidation() {
  currentJob = null
}
```

Reply:
Yes okay it is clear with a new private method, no need for comments in that case.